src/textual/command_palette.py

```diff
@@ -0,0 +1,754 @@
+    ) -> None:
+        """Consume a source of matching commands, feeding the given command queue.
+
+        Args:
+            source: The source to consume.
+            commands: The command queue to feed.
+        """
+        async for hit in source:
+            await commands.put(hit)
+
+    async def _search_for(self, search_value: str) -> CommandMatches:
+        """Search for a given search value amongst all of the command sources.
+
+        Args:
+            search_value: The value to search for.
+
+        Yields:
+            The hits made amongst the registered command sources.
+        """
+
+        # Get the style for highlighted parts of a hit match.
+        match_style = self._sans_background(
+            self.get_component_rich_style("command-palette--highlight")
+        )
+
+        # Set up a queue to stream in the command hits from all the sources.
+        commands: Queue[CommandSourceHit] = Queue()
+
+        # Fire up an instance of each command source, inside a task, and
+        # have them go start looking for matches.
+        searches = [
+            create_task(
+                self._consume(
+                    source(self._calling_screen, match_style).search_for(search_value),
+                    commands,
+                )
+            )
+            for source in self._sources
+        ]
+
+        # Set up a delay for showing that we're busy.
+        self._start_busy_countdown()
+
+        # Now, while there's some task running...
+        while any(not search.done() for search in searches):
+            try:
+                # ...briefly wait for something on the stack. If we get
+                # something yield it up to our caller.
+                yield await wait_for(commands.get(), 0.1)
+            except TimeoutError:
+                # A timeout is fine. We're just going to go back round again
+                # and see if anything else has turned up.
+                pass
+            else:
+                # There was no timeout, which means that we managed to yield
+                # up that command; we're done with it so let the queue know.
+                commands.task_done()
+
+        # Having finished the main processing loop, we're not busy any more.
+        # Anything left in the queue (see next) will fall out more or less
+        # instantly.
+        self._stop_busy_countdown()
+
+        # If all the sources are pretty fast it could be that we've reached
+        # this point but the queue isn't empty yet. So here we flush the
+        # queue of anything left. Note though that rather than busy-spin the
+        # queue and just pull items and yield them, we keep using the
+        # await/wait_for so we don't block until we're done. Not doing this
+        # makes typing into the input *very* choppy when you have very fast
+        # sources.
+        while not commands.empty():
+            try:
+                yield await wait_for(commands.get(), 0.1)
```

Comment:
I don't think we need the `wait_for` here. If the queue is not empty, then `get` is guaranteed to return quickly.

Reply:
See the comment explaining this loop for why the `wait_for` is there. Long story short: it makes it all a lot more responsive when the queue has been flooded.

Reply:
I don't see how. If `commands` is not empty, then `await commands.get()` should return instantly.